### PR TITLE
feat(testing): Introduce node-api E2E testing

### PR DESCRIPTION
**Pull request by calbera**
- add to consensus client https://github.com/attestantio/go-eth2-client
- Still TODO: proof/ namespace manual http client

## Summary by CodeRabbit

- **New Features**
	- Introduced a new API endpoint for node interactions with enhanced port management.
	- Added new command-line options for improved configurability of the beacon node.
	- Implemented a client for interacting with the BeaconKitNode.
	- Created placeholder API functions to maintain compatibility during ongoing development.
	- Expanded the test suite for end-to-end validation of the Beacon API.
	- Defined constants for client configurations and deposit settings in the testing environment.
	- Streamlined the unit fuzz testing process for focused testing.
	- Added debugging capabilities for the BeaconBlock struct during SSZ encoding.

- **Bug Fixes**
	- Updated status representation in the backend process for better clarity.

- **Chores**
	- Added new dependencies for logging, validation, and telemetry to enhance maintainability.
	- Centralized testing configurations for improved management. 

- **Documentation**
	- Comments added to outline future enhancements for structured ID management.

### CODE REVIEW

File: build/scripts/linting.mk [hunk 1]
```diff
@@ -85,7 +85,7 @@ nilaway:
 	for dir in $$dirs; do \
 		count=$$((count + 1)); \
 		printf "[%d/%d modules complete] Running nilaway in %s\n" $$count $$total $$dir && \
-		(cd $$dir && go run go.uber.org/nilaway/cmd/nilaway -exclude-errors-in-files "pkg/components/module,pkg/deposit,pkg/cosmos" -v ./...) || exit 1; \
+		(cd $$dir && go run go.uber.org/nilaway/cmd/nilaway -exclude-errors-in-files "pkg/components/module,pkg/deposit,pkg/cosmos,pkg/eip4788" -v ./...) || exit 1; \
```
Comment (itsdevbear) [inline build/scripts/linting.mk:88]: reason?

File: mod/node-api/client/client.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+import (
+	"context"
+
+	beaconhttp "github.com/attestantio/go-eth2-client/http"
+	"github.com/berachain/beacon-kit/mod/errors"
+)
+
+// New creates a new BeaconKitNode client instance with a given cancel context.
+func New(
```
Comment (itsdevbear) [inline mod/node-api/client/client.go:31]: this should be in testing imo

File: build/scripts/linting.mk (hunk 1 above)
Comment (calbera) [reply build/scripts/linting.mk:88]: Another bindings package similar to "pkg/deposit"

File: mod/node-api/client/client.go (hunk 2 above)
Comment (calbera) [reply mod/node-api/client/client.go:31]: do we want to keep a client implementation anyone can use?

**Review by itsdevbear — CHANGES_REQUESTED** [review]
move client to testing folder otherwise lgtm

File: testing/e2e/e2e_api_test.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+	// Ensure the state root is not nil.
+	stateRoot, err := client.BeaconStateRoot(
+		s.Ctx(),
+		&beaconapi.BeaconStateRootOpts{
+			State: utils.StateIDHead,
+		},
+	)
+	s.Require().NoError(err)
+	s.Require().NotEmpty(stateRoot)
```
Comment (itsdevbear) [inline testing/e2e/e2e_api_test.go:50]: add a check for non 0x00000.00000

**Pull request merged**